### PR TITLE
Ecdh

### PR DESCRIPTION
**Pull request by andreistefanescu**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by andreistefanescu** [comment]
@apetcher-amazon I plan to update the README after your first review

File: SAW/proof/ECDH/ECDH.saw [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+include "../common/helpers.saw";
+include "../common/internal.saw";
+include "../common/memory.saw";
+include "../BN/BN.saw";
+include "../EC/EC.saw";
+
+
+// Specification of the EVP function
+let EVP_PKEY_derive_spec = do {
```
Comment (apetcher-amazon) [inline SAW/proof/ECDH/ECDH.saw:33]: To be consistent with other proofs, should this go in a file called evp_function_specs.saw?

**Review by apetcher-amazon — COMMENTED** [review]
Additional general comments:

- Add verification for the other EVP functions needed for ECDH: EVP_PKEY_derive_init, EVP_PKEY_derive_set_peer
- Add verification for other functions needed to produce a new key pair. This should probably be shared between ECDH and ECDSA proofs. For example: https://wiki.openssl.org/index.php/Elliptic_Curve_Diffie_Hellman#Using_ECDH_in_OpenSSL
- Update README files.

File: SAW/proof/ECDSA/ECDSA.saw [hunk 2]
```diff
@@ -46,6 +36,9 @@ include "../EC/EC.saw";
 include "memory.saw";
 
 
+let max_sig_len = eval_size {| 1 + 1 + 1 + 1 + (ec_bits / 8 + 1) + 1 + 1 + (ec_bits / 8 + 1) |};
```
Comment (apetcher-amazon) [inline SAW/proof/ECDSA/ECDSA.saw:39]: Looks like I missed this one in the ECDSA review. Can you explain the calculation here? Also, see if you can avoid duplication between sig_len and max_sig_len.

**Review by apetcher-amazon — CHANGES_REQUESTED** [review]
Looks good. I just have a couple of small documentation requests.

**Review by apetcher-amazon — APPROVED** [review]

**Pull request merged**